src/main/java/org/folio/rest/exception/HttpException.java

```diff
@@ -17,14 +18,35 @@ public HttpException(int code, String message) {
     super(StringUtils.isNotEmpty(message) ? message : ErrorCodes.GENERIC_ERROR_CODE.getDescription());
     this.code = code;
     this.errors = new Errors()
-      .withErrors(Collections.singletonList(new Error().withCode(ErrorCodes.GENERIC_ERROR_CODE.getCode()).withMessage(message)))
+      .withErrors(singletonList(new Error().withCode(ErrorCodes.GENERIC_ERROR_CODE.getCode()).withMessage(message)))
+      .withTotalRecords(1);
+  }
+
+  public HttpException(int code, String message, Throwable cause) {
+    super(message, cause);
+    this.code = code;
+    Parameter causeParam = new Parameter().withKey("cause").withValue(cause.getMessage());
+    Error error = new Error()
+      .withCode(ErrorCodes.GENERIC_ERROR_CODE.getCode())
+      .withMessage(message)
+      .withParameters(singletonList(causeParam));
```

Comment:
Can we use List.of(), As far as I know singletonList will allow null

Reply:
other places as well

Reply:
I was prefering `singletonList` for single element lists, but I will switch to `List.of`.